src/client/app/reducers/formReducer.js

```diff
@@ -0,0 +1,68 @@
+import {
+    FORM_SUCCESS,
+    SUBMIT_ERROR,
+    SET_FORM_FIELDS,
+    UPDATE_FORM_FIELD,
+    SET_USER_AUTOCOMPLETE
+} from '../constants';
+
+const initialstate = {
+    errors: []
+};
+
+export default function formReducer (state = initialstate, action) {
+    switch (action.type) {
+
+    case SET_FORM_FIELDS:
+        const { fields, modelName, formType } = action.data;
+        return {
+            ...state,
```

Comment:
Oh cool, thanks for explaining! Are there any compatibility concerns with this and the above async plugin? I don't think it's necessary for us to support anything but the latest browsers, but might be good to note in the ReadMe or elsewhere so folks know.